grpc/src/main/java/com/linecorp/armeria/internal/grpc/StatusExceptionConverter.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.grpc;
+
+import com.linecorp.armeria.common.grpc.protocol.ArmeriaStatusException;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+
+public final class StatusExceptionConverter {
+
+    public static StatusRuntimeException toGrpc(ArmeriaStatusException armeria) {
+        StatusRuntimeException converted = Status.fromCodeValue(armeria.getCode())
+                                                 .withDescription(armeria.getMessage())
+                                                 .withCause(armeria.getCause())
+                                                 .asRuntimeException();
+        converted.setStackTrace(armeria.getStackTrace());
```

Comment:
I'm curious if we fill the stack trace twice - 1) at `ArmeriaStatusException.<init>` and 2) `asRuntimeException()` here. If so, can we avoid the 2nd one?

Reply:
Yeah we do... I almost could except that the constructor we'd need on `StatusRuntimeException`, which accepts a boolean for filling the trace, is not `public`. Since this is only in error cases, maybe it's uncommon enough not to worry about the performance, but what do you think?

Other options
- Create an exception that extends `StatusRuntimeException` with a constructor we can use
- Create the `Status` manually and set the cause to the `ArmeriaStatusException`

In both cases, the behavior of the exception would be slightly different than upstream. Quite small difference but lean towards keeping them the same.

Reply:
OK. Let's leave it as it is for now. Could you add something lile:

```
XXX(anuraaga): We fill stack trace twice. Can we avoid the 2nd `fillInStackTrace()`?
```

..?